tests/test_analysis/test_hydrogen_bonding.py

```diff
@@ -0,0 +1,172 @@
+    mock_traj = MagicMock()
+    mock_load_single_traj.return_value = mock_traj
+
+    # Define the expected output from baker_hubbard
+    expected_hbonds = [(1, 2, 3), (4, 5, 6)]
+    mock_baker_hubbard.return_value = expected_hbonds
+
+    # Call the _run method
+    traj_file = "rec0_butane_123456"
+    top_file = "top_sim0_butane_123456"
+    result = baker_hubbard_tool._run(traj_file, top_file)
+
+    # Assertions
+    mock_load_single_traj.assert_called_once_with(
+        baker_hubbard_tool.path_registry, top_file, traj_file
+    )
+    mock_baker_hubbard.assert_called_once_with(
+        mock_traj, 0.1, exclude_water=True, periodic=True, sidechain_only=False
+    )
+    assert result == f"Succeeded. {expected_hbonds}"
+
+
+@patch(
+    "mdagent.tools.base_tools.analysis_tools.hydrogen_bonding_tools.load_single_traj"
+)
+def test_run_fail_baker_hubbard(mock_load_single_traj, baker_hubbard_tool):
+    # Simulate the trajectory loading failure
+    mock_load_single_traj.return_value = None
+
+    # Call the _run method
+    traj_file = "rec0_butane_123456"
+    top_file = "top_sim0_butane_123456"
+    result = baker_hubbard_tool._run(traj_file, top_file)
+
+    # Assertions
+    mock_load_single_traj.assert_called_once_with(
+        baker_hubbard_tool.path_registry, top_file, traj_file
+    )
+    assert result == "Failed. Trajectory could not be loaded."
+
+
+@patch(
+    "mdagent.tools.base_tools.analysis_tools.hydrogen_bonding_tools.load_single_traj"
+)
+@patch("mdtraj.kabsch_sander")
+def test_run_success_kabsch_sander(
+    mock_kabsch_sander, mock_load_single_traj, kabsch_sander_tool
+):
+    # Create a mock trajectory
+    mock_traj = MagicMock()
+    mock_load_single_traj.return_value = mock_traj
+
+    # Define the expected output from kabsch_sander
+    expected_hbonds = ([(0, 1), (2, 3)], [0.5, 0.7])
+    mock_kabsch_sander.return_value = expected_hbonds
+
+    # Call the _run method
+    traj_file = "rec0_butane_123456"
+    top_file = "top_sim0_butane_123456"
+    result = kabsch_sander_tool._run(traj_file, top_file)
+
+    # Assertions
+    mock_load_single_traj.assert_called_once_with(
+        kabsch_sander_tool.path_registry, top_file, traj_file
+    )
+    mock_kabsch_sander.assert_called_once_with(mock_traj)
+    assert result == f"Succeeded. {expected_hbonds}"
+
+
+@patch(
+    "mdagent.tools.base_tools.analysis_tools.hydrogen_bonding_tools.load_single_traj"
+)
+def test_run_fail_kabsch_sander(mock_load_single_traj, kabsch_sander_tool):
+    # Simulate the trajectory loading failure
+    mock_load_single_traj.return_value = None
+
+    # Call the _run method
+    traj_file = "rec0_butane_123456"
+    top_file = "top_sim0_butane_123456"
+    result = kabsch_sander_tool._run(traj_file, top_file)
+
+    # Assertions
+    mock_load_single_traj.assert_called_once_with(
+        kabsch_sander_tool.path_registry, top_file, traj_file
+    )
+    assert result == "Failed. Trajectory could not be loaded."
+
+
+@patch(
```

Comment:
Since you have this many times, can you make it a fixture and just apply the patch by using the fixture?